x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/interceptor/IndicesAliasesRequestInterceptor.java

```diff
@@ -70,7 +70,7 @@ public void intercept(IndicesAliasesRequest request, Authentication authenticati
                                 permissionsMap.computeIfAbsent(alias, userPermissions.indices()::allowedActionsMatcher);
                             if (Operations.subsetOf(aliasPermissions, indexPermissions) == false) {
                                 // TODO we've already audited a access granted event so this is going to look ugly
-                                auditTrailService.accessDenied(authentication, action, request, userPermissions.names());
+                                auditTrailService.accessDenied(null, authentication, action, request, userPermissions.names());
```

Comment:
I prefer us to be explicit here: `threadContext.getTransient(AUDIT_REQUEST_ID)`.
It is a case of *tamper request* , if audit request Id is missing.

Reply:
I think, in this case, it's better to be `null`, because it's already a denial.
I'd rather fail on the _denied_ rather than fail on a tampered request.

Reply:
I've expressed myself poorly. I agree failing with `access_denied` is better and we should not change this.

I meant that since these request interceptors are called after `AuthorizationService#authorize` https://github.com/elastic/elasticsearch/blob/e179bd393dadc5151ea28ca7ec125743a437ca51/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/filter/SecurityActionFilter.java#L174

we can be confident that they are assigned an audit request id already. If they don't have it, then it must be a case of tamper request. Does this sound reasonable?